src/objectExplorer/objectExplorerService.ts

```diff
@@ -244,6 +244,37 @@ export class ObjectExplorerService {
         return [new AddConnectionTreeNode()];
     }
 
+    /**
+     * Removes password from a saved profile
+     */
+    private async removeProfilePassword(connection: IConnectionCredentials): Promise<void> {
+        // if the password is saved in the credential store, remove it
+        let profile = connection as IConnectionProfile;
+        await this._connectionManager.connectionStore.removeProfile(profile);
+        const newProfile = Object.assign({}, profile);
+        newProfile.password = '';
+        newProfile.savePassword = false;
```

Comment:
So let's say I have save password set to true and then connect, but it fails due to some other issue (network issues or whatnot). Isn't this going to wipe out the password and then also make it not save the password that I enter in again? Why are you changing savePassword anyways?

Reply:
It's because we prompt for password only if `savePassword` is true when creating a session - https://github.com/microsoft/vscode-mssql/pull/1512/files#diff-9401975d07f4df26ce1df2d776b8c746R371

So, when you have a disconnected node with an incorrect password, clicking on it to expand would prompt you for password again.